amplipi/streams.py

```diff
@@ -78,7 +78,7 @@ class BaseStream:
   def __init__(self, stype: str, name: str, only_src=None, disabled: bool=False, mock=False):
     self.name = name
     self.disabled = disabled
-    self.proc: Optional[subprocess.Popen] = None
+    self.metadata_proc: Optional[subprocess.Popen] = None
```

Comment:
The first proc is intended to be the main running process, most streams also have a secondary metadata process. To handle streams that don't have a metadata process _is_running() should check the main running process. proc is a pretty generic name there may be a better one.

Reply:
Recent changes to process handling have added code that looks like this for carefully shutting down processes without leaving any zombies. Here's what the spotify equivalent looks like:

```python3
    if self.proc:
      try:
        self.proc.terminate()
        self.proc.communicate(timeout=3)
      except Exception as e:
        print(f"failed to terminate spotify stream: {e}")
        self.proc.kill()
      self.proc = None
    if self.mpris:
      try:
        del self.mpris
      except Exception:
        pass
      self.mpris = None
```